lightning/src/routing/router.rs

```diff
@@ -1742,6 +1742,7 @@ mod tests {
 			funding_txo: Some(OutPoint { txid: bitcoin::Txid::from_slice(&[0; 32]).unwrap(), index: 0 }),
 			channel_type: None,
 			short_channel_id,
+			outbound_scid_alias: None,
```

Comment:
Yeah, I understand that it is expected, but currently none of tests check the case where `short_channel_id` is `None`. I guess the behavior is hidden behind the `ChannelDetails` interface, but `short_channel_id` is public so the interface doesn't prevent us from accidentally using `short_channel_id` instead of `get_outbound_payment_scid`. Not sure if it is worth worrying too much about. I suppose we could have `find_route` take some wrapper around `ChannelDetails` to constrain the interface.

Reply:
My concern is more that `get_route` could be updated somewhere to reference `short_channel_id` instead of calling `get_outbound_payment_scid` without a test breaking. The interface doesn't restrict `get_route` to call the right one.